### PR TITLE
[libmicrohttpd] update and various fixes

### PR DESCRIPTION
**Pull request by Karlson2k**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

Changes are in the individual commits for clarity.
They can be squashed into single commit

### CODE REVIEW

**Comment by Karlson2k** [comment]
It is strange that the whole repo is licensed under MIT/Expat license, while any contribution requires acceptance of separate license.
Ok, my contribution isn't that large so I @microsoft-github-policy-service agree

However, according to the repo license, any person have rights to use it, while according to `Contribution License Agreement` only Microsoft and persons who receive the code from Microsoft (directly or indirectly) have rights to use it. 
Isn't it a legal issue?
How can I check that code, which I receive from 3rd party, is coming from Microsoft?
Doesn't these two licenses conflict with each other?

**Comment by Karlson2k** [comment]
> @Karlson2k please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

It is strange that the whole repo is licensed under MIT/Expat license, while any contribution requires acceptance of separate license.
Ok, my contribution isn't that large so I @microsoft-github-policy-service agree

However, according to the repo license, any person have rights to use it, while according to `Contribution License Agreement` only Microsoft and persons who receive the code from Microsoft (directly or indirectly) have rights to use it. 
Isn't it a legal issue?
How can I check that code, which I receive from 3rd party, is coming from Microsoft?
Doesn't these two licenses conflict with each other?

**Comment by Karlson2k** [comment]
> @Karlson2k please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

It is strange that the whole repo is licensed under MIT/Expat license, while any contribution requires acceptance of separate license.
Ok, my contribution isn't that large so I @microsoft-github-policy-service agree

However, according to the repo license, any person have rights to use it, while according to `Contribution License Agreement` only Microsoft and persons who receive the code from Microsoft (directly or indirectly) have rights to use it. 
Isn't it a legal issue?
How can I check that code, which I receive from 3rd party, is coming from Microsoft?
Doesn't these two licenses conflict with each other?

> @Karlson2k please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

It is strange that the whole repo is licensed under MIT/Expat license, while any contribution requires acceptance of separate license.
Ok, my contribution isn't that large so I 
@microsoft-github-policy-service agree

However, according to the repo license, any person have rights to use it, while according to `Contribution License Agreement` only Microsoft and persons who receive the code from Microsoft (directly or indirectly) have rights to use it. 
Isn't it a legal issue?
How can I check that code, which I receive from 3rd party, is coming from Microsoft?
Doesn't these two licenses conflict with each other?

**Review by Cheney-W — CHANGES_REQUESTED** [review]

File: ports/libmicrohttpd/portfile.cmake [hunk 1]
```diff
@@ -1,11 +1,11 @@
-set(MICROHTTPD_VERSION 0.9.75)
+set(MICROHTTPD_VERSION 0.9.77)
 
 vcpkg_download_distfile(ARCHIVE
     URLS
         "https://ftp.gnu.org/gnu/libmicrohttpd/libmicrohttpd-${MICROHTTPD_VERSION}.tar.gz"
         "https://www.mirrorservice.org/sites/ftp.gnu.org/gnu/libmicrohttpd/libmicrohttpd-${MICROHTTPD_VERSION}.tar.gz"
     FILENAME "libmicrohttpd-${MICROHTTPD_VERSION}.tar.gz"
```
Comment (Cheney-W) [inline ports/libmicrohttpd/portfile.cmake:7]:
```suggestion
vcpkg_download_distfile(ARCHIVE
    URLS
        "https://ftp.gnu.org/gnu/libmicrohttpd/libmicrohttpd-${VERSION}.tar.gz"
        "https://www.mirrorservice.org/sites/ftp.gnu.org/gnu/libmicrohttpd/libmicrohttpd-${VERSION}.tar.gz"
    FILENAME "libmicrohttpd-${VERSION}.tar.gz"
```

**Comment by Karlson2k** [comment]
@Cheney-W Do you want to squash all commits?
Otherwise I don't see how I can remove the `port-version`.
Even with squashed commits it needs at least `"port-version": 2`, correct?

**Comment by Cheney-W** [comment]
Please remove `"port-version": 1,` from `vcpkg.json` because you have updated the `"version":` from `0.9.75` to `0.9.77`.

**Comment by Cheney-W** [comment]
If the port files are changed but the `"version"` is still 0.9.75, then the value of `"port-version"` should add 1. Obviously, you are not in this situation now.

File: ports/libmicrohttpd/portfile.cmake (hunk 1 above)
Comment (Karlson2k) [reply ports/libmicrohttpd/portfile.cmake]: These changes are integrated. Custom `MICROHTTPD_VERSION` was removed and standard `VERSION` used instead.

**Comment by Karlson2k** [comment]
@Cheney-W Seems that you are looking at overrall changes by this PR.
Please check individual commits.
This PR has 10 commits with changes of `vcpkg.json`. Each commit is atomic, resulting `vcpkg.json` is fully functional after each commit.
First commit with version upgrade remove `port-version`, next commit add back `port-version` with value `1` and then aeach next commit bump `port-version` by one.
Let me know if you need to squash all of them into single commit. 
We single commit `port-version` must be removed.

**Comment by dg0yt** [comment]
> This PR has 10 commits with changes of `vcpkg.json`. Each commit is atomic, resulting `vcpkg.json` is fully functional after each commit.

The PR will be squased when merged.

**Comment by Karlson2k** [comment]
@dg0yt OK, clear. I'll update it.

File: ports/libmicrohttpd/vcpkg.json [hunk 2]
```diff
@@ -1,14 +1,18 @@
 {
   "name": "libmicrohttpd",
-  "version": "0.9.75",
-  "port-version": 1,
+  "version": "0.9.77",
+  "port-version": 9,
```
[the remark this answers is not in the document]
Comment (Karlson2k) [reply ports/libmicrohttpd/vcpkg.json]: Fixed in new commit

**Comment by Karlson2k** [comment]
The PR has been updated.

**Review by Cheney-W — APPROVED** [review]

**Pull request merged**